tests/backup/backup_helper.go

```diff
@@ -3591,3 +3592,171 @@ func AdditionalScheduledBackupRequestParams(backupScheduleRequest *api.BackupSch
+	}
+
+	userSchdeulePolicyOwnershipAccessConfigs := make([]*api.Ownership_AccessConfig, 0)
+
+	for _, userID := range userIDs {
+		userSchedulePolicyOwnershipAccessConfig := &api.Ownership_AccessConfig{
+			Id:     userID,
+			Access: accessType,
+		}
+		userSchdeulePolicyOwnershipAccessConfigs = append(userSchdeulePolicyOwnershipAccessConfigs, userSchedulePolicyOwnershipAccessConfig)
+	}
+
+	groupSchedulePolicyOwnershipAccessConfigs := make([]*api.Ownership_AccessConfig, 0)
+
+	for _, groupID := range groupIDs {
+		groupSchedulePolicyOwnershipAccessConfig := &api.Ownership_AccessConfig{
+			Id:     groupID,
+			Access: accessType,
+		}
+		groupSchedulePolicyOwnershipAccessConfigs = append(groupSchedulePolicyOwnershipAccessConfigs, groupSchedulePolicyOwnershipAccessConfig)
+	}
+
+	schedulePolicyOwnershipUpdateReq := &api.SchedulePolicyOwnershipUpdateRequest{
+		OrgId:     orgID,
+		Name:      schedulePolicyName,
+		Ownership: &api.Ownership{Groups: groupSchedulePolicyOwnershipAccessConfigs, Collaborators: userSchdeulePolicyOwnershipAccessConfigs},
+		Uid:       schedulePolicyUid,
+	}
+
+	_, err := backupDriver.UpdateOwnershipSchedulePolicy(ctx, schedulePolicyOwnershipUpdateReq)
+	if err != nil {
+		return fmt.Errorf("failed to update schedule policy: %v", err)
+	}
+	return nil
+}
+
+// UpdateruleOwnership Updates the rule object ownership
+func UpdateRuleOwnership(ruleName string, ruleUid string, userNames []string, groups []string, accessType api.Ownership_AccessType, publicAccess api.Ownership_PublicAccessControl, ctx context.Context) error {
+	log.Infof("UpdateruleOwnership for users %v", userNames)
+	backupDriver := Inst().Backup
+	userIDs := make([]string, 0)
+	groupIDs := make([]string, 0)
+	for _, userName := range userNames {
+		userID, err := backup.FetchIDOfUser(userName)
+		if err != nil {
+			return err
+		}
+		userIDs = append(userIDs, userID)
+	}
+
+	for _, group := range groups {
+		groupID, err := backup.FetchIDOfGroup(group)
+		if err != nil {
+			return err
+		}
+		groupIDs = append(groupIDs, groupID)
+	}
+
+	userRuleOwnershipAccessConfigs := make([]*api.Ownership_AccessConfig, 0)
+
+	for _, userID := range userIDs {
+		userRuleOwnershipAccessConfig := &api.Ownership_AccessConfig{
+			Id:     userID,
+			Access: accessType,
+		}
+		userRuleOwnershipAccessConfigs = append(userRuleOwnershipAccessConfigs, userRuleOwnershipAccessConfig)
+	}
+
+	groupRuleOwnershipAccessConfigs := make([]*api.Ownership_AccessConfig, 0)
+
+	for _, groupID := range groupIDs {
+		groupRuleOwnershipAccessConfig := &api.Ownership_AccessConfig{
+			Id:     groupID,
+			Access: accessType,
+		}
+		groupRuleOwnershipAccessConfigs = append(groupRuleOwnershipAccessConfigs, groupRuleOwnershipAccessConfig)
+	}
+
+	ruleOwnershipUpdateReq := &api.RuleOwnershipUpdateRequest{
+		OrgId:     orgID,
+		Name:      ruleName,
+		Ownership: &api.Ownership{Groups: groupRuleOwnershipAccessConfigs, Collaborators: userRuleOwnershipAccessConfigs, Public: &publicAccess},
```

Comment:
Can we split this up to increase readability?

```
&api.Ownership{
    Groups: groupCloudCredentialOwnershipAccessConfigs,
    Collaborators: userCloudCredentialOwnershipAccessConfigs,
    Public: &publicAccess,
}
```

Reply:
Done